File: sample/src/main/java/io/github/ketzalv/validationedittext/sample/base/BaseFragment.java
package io.github.ketzalv.validationedittext.sample.base;

import android.content.Context;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;

public abstract class BaseFragment extends Fragment {

    private IEvent mEvent;

    @Override
    public void onAttach(@NonNull Context context) {
        super.onAttach(context);
        mEvent = (IEvent) context;
    }

    public abstract String getFragmentTag();

    public IEvent getEvent() {
        return mEvent;
    }

    protected void sendEvent(Event event) {
        if (mEvent != null) {
            mEvent.onEvent(event, null);
        }
    }

    protected void sendEvent(Event event, Object object) {
        if (mEvent != null) {
            mEvent.onEvent(event, object);
        }
    }
}

File: sample/src/main/java/io/github/ketzalv/validationedittext/sample/base/BaseFragment.kt
package io.github.ketzalv.validationedittext.sample.base

import android.content.Context
import androidx.fragment.app.Fragment

abstract class BaseFragment : Fragment() {
    private var mEvent: IEvent? = null

    override fun onAttach(context: Context) {
        super.onAttach(context)
        mEvent = context as IEvent
    }

    abstract val fragmentTag: String?
    val event: IEvent?
        get() = mEvent

    protected fun sendEvent(event: Event?) {
        mEvent?.let {
            it.onEvent(event, null)
        }
    }

    protected fun sendEvent(event: Event?, any: Any?) {
        mEvent?.let {
            it.onEvent(event, any)
        }
    }
}